### ISSUE TITLE
Intermittent error while repairing epochs

### ISSUE THREAD

**Issue by craddm** [autoreject/autoreject#39]
I'm trying to use autoreject on some EEG data recorded on a BioSemi system. Sometimes I get the following warning during processing - it seems to happen during the initial fit and looks like it might be related to interpolation (mne's bem.py, via _interpolate_bads_eeg). Specifically this happens during a "repairing epochs" step. I can tell it's during the initial fit as it says it's repairing half the trials, not all of them.

```
C:\Users\Matt\Anaconda3\envs\NewPython\lib\site-packages\mne\bem.py:975: RuntimeWarning: Mean of empty slice.
  radius_init = radii.mean()
C:\Users\Matt\Anaconda3\envs\NewPython\lib\site-packages\numpy\core\_methods.py:80: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
C:\Users\Matt\Anaconda3\envs\NewPython\lib\site-packages\numpy\core\fromnumeric.py:2889: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\Matt\Anaconda3\envs\NewPython\lib\site-packages\numpy\core\_methods.py:73: RuntimeWarning: invalid value encountered in true_divide
  ret, rcount, out=ret, casting='unsafe', subok=False)
```
I'm calling autoreject as follows.
```
# run autoreject
    thresh_func = partial(compute_thresholds, method='bayesian_optimization', random_state = 342)
    n_interpolates = np.array([1, 4, 32])
    consensus_percs = np.linspace(0, 1.0, 11)
    ar = LocalAutoRejectCV(n_interpolates = n_interpolates,
                           consensus_percs = consensus_percs,
                           thresh_func=thresh_func, verbose='tqdm')
    epochs_ar = []    
    #fit Autoreject model to a subset of epochs, then correct all epochs
    epochs_ar = ar.fit(epochs[::2]).transform(epochs)
```

I use epochs[::2] here as I have 480 (~3s long...) trials per participant, and was trying out various options suggested in the comments on issue #16 to deal with instability of solutions (another issue i'm having...). I've had the same issue with [::3] and [::4].

Note that this seems to be contingent on the random state for compute_thresholds. With random_state = 342, this occurs reliably for one dataset I have, but goes away with random_state =343 for the same dataset. Obviously, if I leave this random, it then happens randomly on other datasets.

**Comment by jasmainak** [comment]
Hmm ... how many sensors do you have in your recording? Maybe `n_interpolates` is too large ... so the sphere fit for interpolation is very bad. In the end, `autoreject` should not select that `n_interpolate` due to the cross validation.

Also are you using the latest version of `autoreject` and do you still see the instability problem with `compute_thresholds`?

**Comment by craddm** [comment]
Hmm, yeah - 64 electrodes. So yes, 32 is a bit...over the top. Anyway, yeah, I'd already tried it with 16 instead of 32, which seemed a more reasonable higher-end, just so happened i'd tried it again with 32 before posting. Same problem. Also same problem with 8 as the highest number. Slightly misunderstood how that parameter works through not properly reading and lack of python familiarity ;) - I'd assumed it was running from 1 up 32 in steps of 4 (which doesn't even make sense as it's not divisible). Is it testing each number in the array sequentially? From the log, the first run and set of folds works:
n_interp:   0%|                                          | 0/3 [00:00<?, ?it/s]

But it fails during the second iteration:
n_interp:  33%|███████████▎                      | 1/3 [00:25<00:50, 25.19s/it]

So it was failing with n_interpolates = 4 - doesn't seem like that should be enough to cause sphere-fitting issues?

I just tried again using epochs[::4], and it worked. So then I tried again with epochs[::2] and with n_interpolates = np.array([1, 2, 4, 16]). It failed again during the second iteration, ~70% of the way through the initial repairing epochs stage:
n_interp:  25%|████████▌                         | 1/4 [00:26<01:18, 26.06s/it] 
...
Repairing epochs:  70%|███████████████▍      | 168/239 [00:02<00:01, 66.75it/s] 

(it's 239 because there are 478 trials in this dataset, same problem with others though). So this suggests it was failing with n_interpolates = 2!

**Comment by jasmainak** [comment]
okay, all this seems a bit mysterious. I'll try to first reproduce the problem locally. If you have any script that you can share, it would be great!

**Comment by jasmainak** [comment]
Also, can you look at the sensor locations by doing `epochs.plot_sensors()` or something like that just to see if nothing is off ...

**Comment by craddm** [comment]
Sensor locations seem fine, they're just the standard Biosemi 64 ones included with MNE. Here's the [epochs file](https://dl.dropboxusercontent.com/u/2115135/Extero_Test-epo.fif) I'm having issues with when running the autoreject code I pasted above. If you still need the pre-processing code I can sort that  and the raw data over the weekend.

**Comment by jasmainak** [comment]
Thanks a lot! I was able to reproduce :) I'll get to it soon
